UnityGLTF/Assets/UnityGLTF/Scripts/Async/AsyncCoroutineHelper.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+namespace UnityGLTF
+{
+    public interface IAsyncCoroutineHelper
+    {
+	   Task RunAsTask(IEnumerator coroutine, string name);
+    }
+
+	public class AsyncCoroutineHelper : MonoBehaviour, IAsyncCoroutineHelper

```

Comment:
What are the responsibilities of this class? It looks like it both lets you "start" a coroutine from any thread, and also "converts" it into a Task. Is that right?